apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/Config.java

```diff
@@ -259,6 +260,43 @@
          * @see org.apache.skywalking.apm.agent.core.logging.core.PatternLogger#DEFAULT_CONVERTER_MAP
          */
         public static String PATTERN = "%level %timestamp %thread %class : %msg %throwable";
+
+        public static class JSON {
+            /**
+             * Key of the AgentName
+             */
+            public static String AGENT_NAME_KEY = "agent_name";
```

Comment:
I don't think these belong to Config.

Reply:
This question is unresolved, who will set these keys? Are these configurable?

Reply:
> This question is unresolved, who will set these keys? Are these configurable?

Yes. It can be configured. They are used as key of the generated json map

Reply:
Technically, yes, my question is more about, why and who will config this.

Reply:
> Technically, yes, my question is more about, why and who will config this.

The application developer can config this provided that they use a different json key set for their application. So unification of these keys may be necessary for them.

Reply:
I think know why they are necessary. I prefer don't provide an over-configurable feature. If someone wants it, they could add them by providing a real use case.

Reply:
> I think know why they are necessary. I prefer don't provide an over-configurable feature. If someone wants it, they could add them by providing a real use case.

I partially agree in the sense that some keys like `agent_name` are rarely used by real application, so it can be just a constant (I can remove it). 

But the others, for example the concepts of "@timestamp", "message", "class" are also adopted by other logging framework. In order to simplify the preference of collectors, I suppose it is necessary to just keep these configurations here. They are very similar to what "pattern" does here.

I think the most users can keep the default settings like we always do.

Reply:
> for example the concepts of "@timestamp", "message", "class" are also adopted by other logging framework.

Please provide the official documentation. If they are there, we could adopt it. But either way, still it is not necessary to make them configurable.

Reply:
The `@timestamp` is used in logstash by default, the source code can be find here, https://github.com/elastic/logstash/blob/0d82bc064ca8184722d159e3e8036ffe1931e666/logstash-core/src/main/java/org/logstash/Event.java#L54

The classic `logback` (which is the default logging framework in Spring Boot) uses the following json layout,

https://github.com/qos-ch/logback-contrib/blob/c68c2e5ddc0ed589b0f1e24bac7b9300a0ea5153/json/classic/src/main/java/ch/qos/logback/contrib/json/classic/JsonLayout.java#L105-L113

I suppose so far this is the only controversial part.

I agree in this PR, we can just keep the layout a constant with some reasonable choices of the default keys. And then wait for the feedback from the community.

I will remove them later.

Reply:
Fixed